Modules/CoordinatorStatus/StatusWriter.cpp

```diff
@@ -0,0 +1,88 @@
+#include "StatusWriter.h"
+
+#include "json.hpp"
+
+#include <iostream>
+
+StandartTextWriter::~StandartTextWriter(){
+	std::cout << std::endl;
+}
+
+void StandartTextWriter::PrintMessage(const std::string& msg)
+{
+	std::cout << msg;
+}
+
+void StandartTextWriter::PrintConnectedToolServer(const std::string& host, int16_t port, uint16_t runningTasks, uint16_t totalThreads)
+{
+	std::cout <<  host << ":" << port << " load:" << runningTasks << "/" << totalThreads << std::endl;
+}
+
+void StandartTextWriter::PrintSummary(int running, int thread, int queued)
+{
+	std::cout << std::endl <<"Total load:" << running << "/" << thread << ", queue: " << queued << std::endl;
+}
+
+void StandartTextWriter::PrintSessions(const std::string& started, int usedThreads)
+{
+	std::cout << "Started " << started << ", used: " << usedThreads << std::endl;
+}
+void StandartTextWriter::PrintTools(const std::set<std::string>& toolIds)
+{
+	std::cout << "\nAvailable tools: ";
+	for (const auto & t : toolIds)
+		std::cout << t << ", ";
+}
+
+
+struct JsonWriter::Impl
+{
+	nlohmann::ordered_json jsonResult;
+};
+
+JsonWriter::JsonWriter() : m_impl(new Impl) {};
+
+JsonWriter::~JsonWriter() {
+	std::cout << m_impl->jsonResult.dump(4) << std::endl;
+}
+
+void JsonWriter::PrintMessage(const std::string&)
+{
+	// stub
+}
+void JsonWriter::PrintConnectedToolServer(const std::string& host, int16_t port, uint16_t runningTasks, uint16_t totalThreads)
+{
+	m_impl->jsonResult["connected_tool_servers"][host]["port"] = port;
+	m_impl->jsonResult["connected_tool_servers"][host]["running_tasks"] = runningTasks;
+	m_impl->jsonResult["connected_tool_servers"][host]["total_threads"] = totalThreads;
```

Comment:
I feel that using a thirdparty library for writing just a couple lines of JSON is a huge overshoot. For parsing - yes, I won't mind, but adding a few quotes/braces - I feel like too much.

Reply:
```
	auto& srv = m_impl->jsonResult["connected_tool_servers"][host];
	srv["port"] = port;
	srv["running_tasks"] = runningTasks;
	srv["total_threads"] = totalThreads;
```
Is it better or am I misunderstanding?

Reply:
Yes, I mean that using json library to format lines few JSON lines is overshoot. You need to remove that dependency. 
`os << R"("connected_tool_servers": {")" << host << R"(": { "port":)" << port `
and so on.

Reply:
Why for reinventing the wheel?
This will be creating troubles in future for reading and supporting.
Will also have to control the correctness of the json format yourself
Afterwards it is also planned to expand the functionality for more detailed out, and this solution will force us "to place brackets manually".
I cannot insist, this project is yours, but this solution seems reasonable to me.

Reply:
Ok, sold for "future extensibility". You can leave a json library.